calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java

```diff
@@ -146,7 +146,8 @@
 	private static final String STATE_SCHEDULE_IMPORT = "scheduleImport";
 	private static final String CALENDAR_INIT_PARAMETER = "calendar";
 	private static final int HOURS_PER_DAY = 24;
-	private static final int NUMBER_HOURS_PER_PAGE_FOR_WEEK_VIEW = 10;
+	private static final int NUMBER_HOURS_PER_PAGE = ServerConfigurationService.getInt("calendar.hoursPerPage", 10);
```

Comment:
As you mention > 24 produces bad results, so I think this should have an upper restriction if it's > 24 just set it to 24. Should probably also have a lower restriction of 1.

Reply:
You'd probably have to limit it in the constructor though

Reply:
I agree with you about the upper restriction.
About the lower restriction, < 8 produces that some hours can't be viewed in every page (since start hours are 0:00, 8:00 and 14:00). And < 10 produces hours not viewable in the third frame (14:00-0:00).
So a lower restriction of 10 (the default value in fact) does make sense at least while page start hours can't be configured, doesn't it?

Reply:
Sounds good, I just didn't know what the lower limit was. Thanks!